unix_integration/resolver/src/unix_config.rs

```diff
@@ -14,9 +14,68 @@ use serde::Deserialize;
 
 use kanidm_unix_common::constants::*;
 
+// This bit of magic lets us deserialise the old config and the new versions.
+
+#[derive(Debug, Deserialize)]
+#[serde(untagged)]
+enum ConfigUntagged {
+    Versioned(ConfigVersion),
+    Legacy(ConfigInt),
+}
+
+#[derive(Debug, Deserialize)]
+#[serde(tag = "version")]
+enum ConfigVersion {
+    #[serde(rename = "2")]
+    V2 {
+        #[serde(flatten)]
+        values: ConfigV2,
+    },
+}
+
+#[derive(Debug, Deserialize)]
+struct ConfigV2 {
+    cache_db_path: Option<String>,
+    sock_path: Option<String>,
+    task_sock_path: Option<String>,
+
+    cache_timeout: Option<u64>,
+
+    default_shell: Option<String>,
+    home_prefix: Option<String>,
+    home_mount_prefix: Option<String>,
+    home_attr: Option<String>,
+    home_alias: Option<String>,
+    use_etc_skel: Option<bool>,
+    uid_attr_map: Option<String>,
+    gid_attr_map: Option<String>,
+    selinux: Option<bool>,
+
+    hsm_pin_path: Option<String>,
+    hsm_type: Option<String>,
+    tpm_tcti_name: Option<String>,
+
+    kanidm: Option<KanidmConfigV2>,
+}
+
+#[derive(Debug, Deserialize)]
+pub struct GroupMap {
+    pub local: String,
+    pub with: String,
+}
+
+#[derive(Debug, Deserialize)]
+struct KanidmConfigV2 {
+    conn_timeout: Option<u64>,
+    request_timeout: Option<u64>,
+    pam_allowed_login_groups: Option<Vec<String>>,
+    extend: Vec<GroupMap>,
+}
+
 #[derive(Debug, Deserialize)]
 struct ConfigInt {
-    db_path: Option<String>,
+    #[serde(alias = "db_path")]
```

Comment:
why bother aliasing?

Reply:
It's so old config files will still load.

Reply:
But that is the old config. The new config is `ConfigV2`. Could have just kept the old name since that is the actual name of the old config.

Reply:
I renamed the value before I realised that actually I needed to make a whole separate config version to achieve what I needed. So I think we can either revert it back or leave the alias, it's not really a big deal either way.